code_search/kubeflow/components/params.libsonnet

```diff
@@ -68,27 +68,36 @@
       modelPath: $.global.t2tWorkingDir + "/output/export/Servo",
       modelServerImage: "gcr.io/kubeflow-images-public/tensorflow-serving-1.8:latest",
       cloud: "gcp",
-      gcpCredentialSecretName: "gcp-credentials",
+      gcpCredentialSecretName: "user-gcp-sa",
     },
 
     "nmslib": {
-      name: null,
       replicas: 1,
-      image: "gcr.io/kubeflow-dev/code-search:v20180621-266e689",
+      image: "gcr.io/kubeflow-dev/code-search-ui:v20180806-7b0fcaa",
 
-      dataFile: null,
-      indexFile: null,
-      problem: null,
-      dataDir: null,
-      servingUrl: null,
+      problem: "null",
+      dataDir: "null",
+      lookupFile: "null",
+      indexFile: "null",
+      servingUrl: "null",
     },
 
-    "nms-creator": {
-      name: "nms-creator",
+    "search-index-creator": {
+      name: "search-index-creator",
+
+      dataDir: $.global.t2tWorkingDir + "/data",
+      lookupFile: $.global.t2tWorkingDir + "/code_search_index.csv",
+      indexFile: $.global.t2tWorkingDir + "/code_search_index.nmslib",
     },
 
-    "nms-server": {
-      name: "nms-server",
+    "search-index-server": {
+      name: "search-index-server",
+
+      problem: "github_function_docstring",
+      dataDir: $.global.t2tWorkingDir + "/data",
```

Comment:
Do you want dataDir and lookupFile,... to always be a function of t2tWorkingDir or do you want to let users override the parameters in a way that breaks that pattern.

Reply:
It is certainly possible to break this pattern but I wanted to minimize the amount of parameters the user was changing. This allows the users to keep everything under one umbrella, easier to see all artifacts in one place (one GCS directory). The advanced users might go ahead and break this pattern, and should be easy enough once they understand the mechanics.